cinder/volume/api.py

```diff
@@ -799,9 +806,40 @@ def initialize_connection(self, context, volume, connector):
             msg = _("The volume connection cannot be initialized in "
                     "maintenance mode.")
             raise exception.InvalidVolume(reason=msg)
-        init_results = self.volume_rpcapi.initialize_connection(context,
-                                                                volume,
-                                                                connector)
+
+        def _migrate_and_initialize_connection():
+            volume_type = {}
+            if volume.volume_type_id:
+                volume_type = volume_types.get_volume_type(
+                    context.elevated(), volume.volume_type_id)
+            request_spec = {
+                'volume_properties': volume,
+                'volume_type': volume_type,
+                'volume_id': volume.id}
+            try:
+                dest = self.scheduler_rpcapi.find_backend_for_connector(
+                    context, connector, request_spec)
+            except exception.NoValidBackend:
+                LOG.exception("The current backend couldn't be used with"
```

Comment:
I think `LOG.error` would be better. It will still print the `NoValidBackend` exception context, I think, because that's how `oslo.log` works.

Reply:
Also need to add spaces at the end of each quote.

Reply:
This piece of change is outdated, I already updated the LOG and the ending space as well. Marking this as resolved.